lib/Tool/Sorter.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Tool;
+
+use Pimcore\Bundle\AdminBundle\GDPR\DataProvider\DataProviderInterface;
+use Pimcore\Cache\Core\CacheQueueItem;
+use Pimcore\HttpKernel\BundleCollection\ItemInterface;
+use Pimcore\Model\Element\Tag;
+use Pimcore\Model\Staticroute;
+use Pimcore\Targeting\Model\TargetGroupAssignment;
+use Pimcore\Workflow\WorkflowConfig;
+
+/**
+ * Static callbacks for array sorting functions.
+ *
+ * @see https://www.php.net/manual/en/array.sorting.php
+ */
+class Sorter
+{
+    public static function date(array $a, array $b): int
+    {
+        return $a['date'] <=> $b['date'];
+    }
+
+    public static function index(array $a, array $b): int
+    {
+        return $a['index'] <=> $b['index'];
+    }
+
+    public static function name(array $a, array $b): int
+    {
+        return $a['name'] <=> $b['name'];
+    }
+
+    public static function population(array $a, array $b): int
+    {
+        return $a['population'] <=> $b['population'];
+    }
+
+    public static function position(array $a, array $b): int
+    {
+        return $a['position'] <=> $b['position'];
+    }
+
+    public static function priority(array $a, array $b): int
+    {
+        return $a['priority'] <=> $b['priority'];
+    }
+
+    public static function sortIndex(array $a, array $b): int
+    {
+        return $a['sortIndex'] <=> $b['sortIndex'];
+    }
```

Comment:
Those could be replaced by a ["higher-order" function](https://en.wikipedia.org/wiki/Higher-order_function): 

```suggestion
    public static function byArrayKey(string $name): \Closure
    {
        return static fn (array $a, array $b): int => $a[$name] <=> $b[$name];
    }
```

The usage would be like: `usort($data, Sorter::byArrayKey('date'))`

Reply:
I wanted to avoid higher-order functions because it would result in additional overhead. The PHP array sorter functions are really smart because they will not even attempt to invoke the closure if the input array is fewer than two items.

But I do agree that this results in neater code.